py/desispec/scripts/extract.py

```diff
@@ -170,7 +170,12 @@ def main(args):
         dw = 0.7
 
     if args.barycentric_correction :
-        barycentric_correction_factor = barycentric_correction_multiplicative_factor(img.meta)
+        try:
```

Comment:
Personally I'd prefer an "if" statement here. If the key is in img.meta, else use fibermap.meta. Better yet, check both with if-elif-else where we return a more informative error in the else.

Try's are great for robustness but since we know the failure point here I think it's more efficient/readable to address it directly

Reply:
I originally used try/except because it was fewer lines of code, but agree that if/else is *clearer* code, and that is valuable.  I switched it.